packages/admin/cms-admin/src/form/file/FinalFormFileUpload.tsx

```diff
@@ -5,25 +5,19 @@ import { FieldRenderProps } from "react-final-form";
 import { FormattedMessage } from "react-intl";
 
 import { useCmsBlockContext } from "../../blocks/useCmsBlockContext";
+import { GQLFileUpload } from "../../graphql.generated";
 import { GQLFinalFormFileUploadFragment } from "./FinalFormFileUpload.generated";
 
 export const finalFormFileUploadFragment = gql`
     fragment FinalFormFileUpload on FileUpload {
         id
         name
         size
+        downloadUrl
     }
 `;
 
-type SuccessfulApiResponse = {
-    id: string;
-    createdAt: string;
-    updatedAt: string;
-    name: string;
-    size: number;
-    mimetype: string;
-    contentHash: string;
-};
+type SuccessfulApiResponse = GQLFileUpload & { downloadUrl: string };
```

Comment:
why doesn't it contain downloadUrl?

and why don't you use the fragment type?

Reply:
> why doesn't it contain downloadUrl?

It does. This was probably an oversight due to refactorings.

> and why don't you use the fragment type?

Because it's not the GraphQL data loaded, but the response of the upload endpoint. Using the base GraphQL type isn't correct either. I reverted this change: cc42f6c5bbb753b8e1c58cec93097754202b7cc6